src/engine/strat_engine/pool.rs

```diff
@@ -127,6 +143,7 @@ impl Pool for StratPool {
         let mut bds = try!(initialize(&self.pool_uuid, devices, MIN_MDA_SIZE, force));
         let bdev_paths = bds.iter().map(|p| p.1.devnode.clone()).collect();
         self.block_devs.append(&mut bds);
+        try!(self.write_metadata());
```

Comment:
Should you be doing the same for ```add_cachedevs```? Making ```add_cachedevs``` unimplemented would be fine with me, too.